sdk/core/core-util/src/checkEnvironment.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+declare global {
+  interface Window {
+    document: unknown;
+  }
+
+  interface DedicatedWorkerGlobalScope {
+    constructor: {
+      name: string;
+    };
+  }
+
+  interface Navigator {
+    product: string;
+  }
+
+  interface DenoGlobal {
+    version: {
+      deno: string;
+    };
+  }
+
+  interface BunGlobal {
+    version: string;
+  }
+
+  // eslint-disable-next-line @azure/azure-sdk/ts-no-window
+  const window: Window;
+  const self: DedicatedWorkerGlobalScope;
+  const Deno: DenoGlobal;
+  const Bun: BunGlobal;
+  const navigator: Navigator;
+}
```

Comment:
Can we put these globals in an ambient .d.ts so they don't end up populating downstream?

Reply:
They're not populating the top level `core-util.d.ts` rollup